stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ListenerUtils.java

```diff
@@ -0,0 +1,267 @@
+        if (service != null) {
+            try {
+                declareQueueIfNotExists(service, channel);
+            } catch (IOException e) {
+                return RabbitMQUtils.returnErrorValue("I/O Error occurred while declaring the queue: " +
+                        e.getMessage());
+            }
+            if (isStarted()) {
+                services =
+                        (ArrayList<ObjectValue>) listenerObjectValue.getNativeData(RabbitMQConstants.CONSUMER_SERVICES);
+                startReceivingMessages(service, channel, listenerObjectValue);
+            }
+            services.add(service);
+            listenerObjectValue.addNativeData(RabbitMQConstants.CONSUMER_SERVICES, services);
+        }
+        return null;
+    }
+
+    public static Object start(ObjectValue listenerObjectValue) {
+        Strand strand = Scheduler.getStrand();
+        boolean autoAck;
+        ObjectValue channelObject = (ObjectValue) listenerObjectValue.get(RabbitMQConstants.CHANNEL_REFERENCE);
+        Channel channel = (Channel) listenerObjectValue.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+        ArrayList<ObjectValue> services =
+                (ArrayList<ObjectValue>) listenerObjectValue.getNativeData(RabbitMQConstants.CONSUMER_SERVICES);
+        @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+        ArrayList<ObjectValue> startedServices =
+                (ArrayList<ObjectValue>) listenerObjectValue.getNativeData(RabbitMQConstants.STARTED_SERVICES);
+        if (services != null && !services.isEmpty()) {
+            for (ObjectValue service : services) {
+                if (startedServices == null || !startedServices.contains(service)) {
+                    MapValue serviceConfig =
+                            (MapValue) service.getType().getAnnotation(RabbitMQConstants.PACKAGE_RABBITMQ,
+                                    RabbitMQConstants.SERVICE_CONFIG);
+                    @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+                    MapValue<String, Object> queueConfig =
+                            (MapValue<String, Object>) serviceConfig.getMapValue(RabbitMQConstants.ALIAS_QUEUE_CONFIG);
+                    autoAck = getAckMode(service);
+                    boolean isQosSet = channelObject.getNativeData(RabbitMQConstants.QOS_STATUS) != null;
+                    if (!isQosSet) {
+                        try {
+                            handleBasicQos(channel, queueConfig);
+                        } catch (RabbitMQConnectorException exception) {
+                            return RabbitMQUtils.returnErrorValue("Error occurred while setting the QoS settings."
+                                    + exception.getDetail());
+                        }
+                    }
+                    MessageDispatcher messageDispatcher = new MessageDispatcher(service, channel, autoAck, runtime, strand);
```

Comment:
Here `runtime` could be null. Since this is a public API, we can't rely or the order of invocation.